### PR TITLE
Fixeswarnings in vsc

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

Fixes the current linter warnings that pop up now.
Note this has not been tested except for the one case where i was unsure but there should be no behavior changes.
Except maybe a behavior change for the uplink that non constant condition probably never actually worked.
Some of the stuff that i had to touch here is poorly executed anyway but i honestly cannot be bothered to completly rewrite all of this code.

## Why It's Good For The Game

Allows coder to focus on their code and not have unrelated warnings pop up.

</details>

## Changelog
:cl:
fix: Fixes non constant expressions in switch functions
fix: Fixes missing trigger for possible input values in switch functions (added else)
fix: Replaces || inside if conditions for the switch functions with ,

/:cl:

### CODE REVIEW

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -106,6 +106,8 @@
 				newletter += "[newletter]"
 			if(20)
 				newletter += "[newletter][newletter]"
+			else
+				newletter += ""
```
Comment (ike709) [inline code/modules/mob/mob_helpers.dm:110]:
```suggestion
				continue
```

**Comment by MNarath1** [comment]
This way of handling it looks much cleaner than using if so i think its better

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -99,12 +99,12 @@
 				newletter = "...huuuhhh..."
 			else if(newletter == ".")
 				newletter = " *BURP*."
-		switch(rand(1, 20))
+		switch(prob(10) ? rand(1, 3) : 0)
```
Comment (ike709) [inline code/modules/mob/mob_helpers.dm]:
```suggestion
		switch(prob(15) ? rand(1, 3) : 0)
```
Unless my math is wrong, 3 valid numbers in 20 is 15% that it'll be a valid number.

File: code/modules/surgery/advanced/lobotomy.dm [hunk 3]
```diff
@@ -44,7 +44,7 @@
 	target.cure_all_traumas(TRAUMA_RESILIENCE_LOBOTOMY)
 	if(target.mind && target.mind.has_antag_datum(/datum/antagonist/brainwashed))
 		target.mind.remove_antag_datum(/datum/antagonist/brainwashed)
-	switch(rand(1,4))//Now let's see what hopefully-not-important part of the brain we cut off
+	switch(prob(75) ? rand(1,3) : 0)//Now let's see what hopefully-not-important part of the brain we cut off
```
Comment (ike709) [inline code/modules/surgery/advanced/lobotomy.dm]:
```suggestion
	switch(rand(0,3))//Now let's see what hopefully-not-important part of the brain we cut off
```

File: code/modules/vending/_vending.dm [hunk 4]
```diff
@@ -427,16 +427,16 @@ GLOBAL_LIST_EMPTY(vending_products)
 	else
 		. = ..()
 		if(tiltable && !tilted && I.force)
-			switch(rand(1, 100))
-				if(1 to 5)
+			switch(prob(44) ? rand(1, 100) : 0)
```
Comment (ike709) [inline code/modules/vending/_vending.dm]:
```suggestion
			switch(prob(44) ? rand(1, 10) : 0)
```

Comment (ike709) [reply code/modules/vending/_vending.dm]: I'm not sure how well your probabilities line up (cba to do the math) but it seems probably fine.

File: code/modules/mob/mob_helpers.dm (hunk 2 above)
Comment (MNarath1) [reply code/modules/mob/mob_helpers.dm]: ironically i had that and then changed it to 10 percent

**Pull request merged**